File: Atcoder/ABC/189/C.cpp
#include <bits/stdc++.h>
using namespace std ;
using V_2 = vector<vector<int>> ;
using ll = long long ;
#define INF (ll)1e8

int n ;

int main(){
  cin >> n ;
  int A[n] ;
  for(int i = 0 ; i < n ; i++) cin >> A[i] ;
  
  int res = 0 , tmp ;
  for(int i = 0 ; i < n ; i++){
    tmp = INF ;
    for(int j = i ; j < n ; j++){
      tmp = min(tmp,A[j]) ;
      res = max(res,(j - i + 1) * tmp) ;
    }
  }
  cout << res << endl ;
}

File: Atcoder/ABC/189/C.py
import numpy as np

n = int(input())
A = list(map(int,input().split()))
A = np.array(A)
tmp = 0
for i in range(0,n) :
  res = 1000008
  for j in range(i,n) :
    res = min(res,A[j])
    tmp = max(tmp,(j-i+1)*res)
print(tmp)